talk/basicconcepts/coresyntax.tex

```diff
@@ -47,51 +47,73 @@
   \begin{cppcode}
     bool b = true;            // boolean, true or false
 
-    char c = 'a';             // 8 bit ASCII char
+    char c = 'a';         // min 8 bit integer (signed or not)
+                              // can store an ASCII character
+    signed char c = 4;        // min 8 bit signed integer
+    unsigned char c = 4;      // min 8 bit unsigned integer
+
     char* s = "a C string";   // array of chars ended by \0
     string t = "a C++ string";// class provided by the STL
 
-    char c = -3;              // 8 bit integer (signed or not)
-    unsigned char c = 4;      // 8 bit unsigned integer
-
-    short int s = -444;       // 16 bit signed integer
-    unsigned short s = 444;   // 16 bit unsigned integer
+    short int s = -444;       // min 16 bit signed integer
+    unsigned short s = 444;   // min 16 bit unsigned integer
     short s = -444;           // int is optional
   \end{cppcode}
 \end{frame}
 \begin{frame}[fragile]
   \frametitlecpp[98]{Basic types(2)}
   \begin{cppcode}
-    int i = -123456;          // 32 bit signed integer
-    unsigned int i = 1234567; // 32 bit signed integer
+    int i = -123456;          // min 16, usually 32 bit
+    unsigned int i = 1234567; // min 16, usually 32 bit
 
-    long l = 0L               // 32 or 64 bits (ptr size)
-    unsigned long l = 0UL;    // 32 or 64 bits (ptr size)
+    long l = 0L               // min 32 bit
+    unsigned long l = 0UL;    // min 32 bit
 
-    long long ll = 0LL;       // 64 bit signed integer
-    unsigned long long l = 0ULL; // 64 bit unsigned integer
+    long long ll = 0LL;          // min 64 bit
+    unsigned long long l = 0ULL; // min 64 bit
 
     float f = 1.23f;          // 32 (23+7+1) bit float
     double d = 1.23E34;       // 64 (52+11+1) bit float
+    long double ld = 1.23E34L // min 64 bit float
```

Comment:
This is actually not exact, because C++ does AFAIK not mandate IEEE 754 FP formats. A C++ implementations could also choose unums or postis for FP types.